toolkits/google/arcade_google/tools/utils.py

```diff
@@ -82,16 +82,17 @@ def parse_email(email_data: dict[str, Any]) -> Optional[dict[str, str]]:
         Optional[Dict[str, str]]: Parsed email details or None if parsing fails.
     """
     try:
-        payload = email_data["payload"]
-        headers = {d["name"].lower(): d["value"] for d in payload["headers"]}
+        payload = email_data.get("payload", {})
+        headers = {d["name"].lower(): d["value"] for d in payload.get("headers", [])}
 
         body_data = _get_email_body(payload)
 
         return {
             "id": email_data.get("id", ""),
+            "thread_id": email_data.get("threadId", ""),
             "from": headers.get("from", ""),
             "date": headers.get("date", ""),
-            "subject": headers.get("subject", "No subject"),
+            "subject": headers.get("subject", ""),
```

Comment:
Good change, ideally we don't insert many of these "interpretations" ourselves 👍